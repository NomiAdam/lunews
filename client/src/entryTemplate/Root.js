import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { isMobile } from 'react-device-detect';
import App from '../App';
import UserProvider from '../context/userContext';
import UiProvider from '../context/uiContext';
import GroupProvider from '../context/groupContext';
import FilterProvider from '../context/filterContext';
import { desktopRoutes, mobileRoutes } from '../routes/routes';

const Root = () => (
    <BrowserRouter>
        <FilterProvider>
            <UserProvider>
                <UiProvider>
                    <GroupProvider>
                        <App routes={ isMobile ? mobileRoutes : desktopRoutes } />
                    </GroupProvider>
                </UiProvider>
            </UserProvider>
        </FilterProvider>
    </BrowserRouter>
);

export default Root;
