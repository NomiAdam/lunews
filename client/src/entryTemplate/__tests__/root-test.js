/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Root from '../Root';

describe('Root', () => {
    it('should be defined', () => {
        expect(Root).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Root routes={ [] } />)).toMatchSnapshot();
    });
});
