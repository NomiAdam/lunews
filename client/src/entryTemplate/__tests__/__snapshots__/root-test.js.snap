// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should match the snapshot 1`] = `
<BrowserRouter>
  <Provider>
    <App
      routes={Array []}
    />
  </Provider>
</BrowserRouter>
`;
